src/java.base/share/classes/java/lang/Integer.java

```diff
@@ -514,8 +548,8 @@ static int getChars(int i, int index, byte[] buf) {
             q = i / 100;
             r = (q * 100) - i;
             i = q;
-            buf[--charPos] = DigitOnes[r];
-            buf[--charPos] = DigitTens[r];
+            charPos -= 2;
+            Unsafe.getUnsafe().putShortUnaligned(buf, Unsafe.ARRAY_BYTE_BASE_OFFSET + charPos, DigitPacks[r]);
```

Comment:
```suggestion
            Unsafe.getUnsafe().putShortUnaligned(buf, Unsafe.ARRAY_BYTE_BASE_OFFSET + charPos, DigitPacks[r], false);
```
You should explicitly specify the endianness, which is platform dependent.